chapters/05-comparison.Rmd

```diff
@@ -27,170 +27,174 @@ distributions_to_evaluate = [
+```{python compareZfitKDEpyRuntimePDFGPU, echo=FALSE, eval=TRUE, fig.cap="Runtime difference of the evaluation phase between the newly proposed algorithms 'Binned', 'FFT', 'ISJ' and the FFT based implementation in KDEpy (run on GPU)", fig.align='center', out.width='100%'}
 figure, axes = runner.plot_runtimes(unrestricted_runtimes, distributions_to_evaluate, ['ZfitBinned', 'ZfitFFT', 'ZfitISJ', 'KDEpyFFT'], 'pdf')
 figure
 ```
 
-The runtime of the PDF evaluation step does not differ much from the one seen on the CPU. All new methods are evaluated in near constant time.
+The runtime of the PDF evaluation phase does not differ much from the one seen on the CPU. All new methods are evaluated in near constant time (figure \@ref(fig:compareZfitKDEpyRuntimePDFGPU)).
 
-```{python compareZfitKDEpyRuntimeTotal, echo=FALSE, eval=TRUE, fig.cap="Runtime difference of the total calculation (instantiation and evaluation step) between the newly proposed algorithms 'Binned', 'FFT', 'ISJ' and the FFT based implementation in KDEpy (run on GPU)", fig.align='center', out.width='100%'}
+```{python compareZfitRuntimePDFGPUwithoutKDEpy, echo=FALSE, eval=TRUE, fig.cap="Runtime difference of the evaluation phase between the newly proposed algorithms 'Binned', 'FFT', 'ISJ' only (run on GPU)", fig.align='center', out.width='100%'}
+figure, axes = runner.plot_runtimes(unrestricted_runtimes, distributions_to_evaluate, ['ZfitBinned', 'ZfitFFT', 'ZfitISJ'], 'pdf')
+figure
+```
+
+Looking at the evaluation runtimes of only the new methods, we can see again that the differences are minimal (figure \@ref(fig:compareZfitRuntimePDFGPUwithoutKDEpy)). Although the binned method does not substantially increase for larger sample sizes, as was seen while running the methods on the CPU (figure \@ref(fig:compareZfitRuntimePDFwithoutKDEpy)).
+
+```{python compareZfitKDEpyRuntimeTotal, echo=FALSE, eval=TRUE, fig.cap="Runtime difference of the total calculation (instantiation and evaluation phase) between the newly proposed algorithms 'Binned', 'FFT', 'ISJ' and the FFT based implementation in KDEpy (run on GPU)", fig.align='center', out.width='100%'}
 figure, axes = runner.plot_runtimes(unrestricted_runtimes, distributions_to_evaluate, ['ZfitBinned', 'ZfitFFT', 'ZfitISJ', 'KDEpyFFT'], 'total')
 figure
 ```
 
-For larger datasets ($n \geq 10^8$) even the total runtime (instantiation and PDF evaluation combined) is faster than KDEpy's FFT method.
\ No newline at end of file
+
+For larger datasets ($n \geq 10^8$) even the total runtime (instantiation and PDF evaluation combined) of the newly proposed binned and FFT methods is slower than for KDEpy's FFT method, i.e these new methods based on TensorFlow and zfit can outperform KDEpy if run on the GPU (figure \@ref(fig:compareZfitKDEpyRuntimeTotal)). 
```

Comment:
As the last thing to basically change: we need a conclusion here, something that answers the questions of "when to use which implementation" or "when is TF better, when KDEpy". The findings of the thesis. And they can be taken from the conclusions, the latter can be shortened and some stuff moved up here.

The findings is all what a thesis is about in the end. So add them here ;) The whole section still tends to be more a description of plots (but also contains conlcusions and reasoning, this is good) and just misses this overall picture and what we see in it.

Or: it seems more like a listing of things. Thinks of it as a presentation of the conclusions (at least for the end of the runtime. It is somewhat mixed with the conclusion session, but the runtime section should be closed within itself and present somewhat findings.

For example, don't write just "can outperform". Say when. Say if there is a tradeoff. Just some sharper conclusions seem missing here.

Do you agree?